derive_builder/examples/validator_fn.rs

```diff
@@ -0,0 +1,37 @@
+//! This example illustrates the use of `validator` to add a pre-build validation
+//! step.
+
+#[macro_use]
+extern crate derive_builder;
+
+#[derive(Builder, Debug, PartialEq)]
+#[builder(build_fn(validator="LoremBuilder::validate"))]
```

Comment:
It doesn't have to be a static function on the builder. A free function is completely valid, including a generic free function - a generic function with trait bounds gets really compelling. I've used that before in my serialization code, so I'm deliberately aligning this with how [Serde](https://serde.rs/attributes.html#field-attributes) referencing functions in attributes.

You could write `validator="Self::validate"` to avoid duplicating the method name if you wanted.

Changing this to ident would prevent extending it in the future, as there's no way to differentiate `"validate"` the struct-local `Ident` from `"validate"` the in-scope free function.